pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/bestpractices/UseCollectionIsEmptyRule.java

```diff
@@ -112,11 +113,14 @@ private String getVariableNameBySuffix(ASTPrimaryExpression primaryExpression) {
     }
 
     private ASTClassOrInterfaceType getTypeOfVariableByName(String varName, ASTPrimaryExpression expr) {
-        ASTClassOrInterfaceBody classBody = expr.getFirstParentOfType(ASTClassOrInterfaceBody.class);
-        List<ASTVariableDeclarator> varDeclarators = classBody.findDescendantsOfType(ASTVariableDeclarator.class);
+        Node classOrEnumBody = expr.getFirstParentOfType(ASTClassOrInterfaceBody.class);
+        if (classOrEnumBody == null) {
```

Comment:
Is it the only case when `classOrEnumBody` can be null? Unfortunately I don't know.

Reply:
Technically it could also be in an annotation type, but I think it's ok, as annotations may not contain code (at least, not code that this rule would pick up on).

Reply:
Technically it could also be in an annotation type, but I think it's ok, as annotations may not contain code (at least, not code that this rule would pick up on).

Reply:
I'll add a test case when merging